airflow/models/dag.py

```diff
@@ -236,11 +237,16 @@ class DAG(LoggingMixin):
     :param dag_id: The id of the DAG; must consist exclusively of alphanumeric
         characters, dashes, dots and underscores (all ASCII)
     :param description: The description for the DAG to e.g. be shown on the webserver
+    :param schedule: Defines the rules according to which DAG runs are scheduled. Can
+        accept cron string, timedelta object, Timetable, or list of Dataset objects.
+        See also :doc:`/howto/timetable`.
     :param schedule_interval: Defines how often that DAG runs, this
         timedelta object gets added to your latest task instance's
-        execution_date to figure out the next schedule
+        execution_date to figure out the next schedule.
+        Note: deprecated in Airflow 2.4; use `schedule` instead.
     :param timetable: Specify which timetable to use (in which case schedule_interval
         must not be set). See :doc:`/howto/timetable` for more information
+        Note: deprecated in Airflow 2.4; use `schedule` instead.
```

Comment:
Should we just remove these entirely from the docstring? New users don’t need to know about them, and those who are using them don’t need to read the docs on them anyway.

Reply:
it is tempting. _however_, i think it 's better to leave like this because a user who encounters them might look to the docstring to understand what they are, and won't be able to easily find out that they are deprecated params.  they might go on a bit of a wild goose chase to find out.

Reply:
wdyt?

Reply:
One approach used elsewhere is to document the existence of deprecated arguments in the new replacement (i.e. `schedule`) and say something like _hey if you see `schedule_interval` or `timetable` they are deprecated forms and synonym of this_. This is slightly more difficult here since `DAG` has so many arguments, but would probably still be good enough.

Reply:
well i'll defer to your judgment....  i.e. if you say let's remove it, i'm happy to do... just... from my perspective it seems odd.

Reply:
Maybe better than what i've done currently is at least to put the deprecation warning *before* the description... anyway... lmk